### ISSUE TITLE
[Roadmap] Optimize and support new features for FSDP

### ISSUE THREAD

**Issue by PopSoda2002** [THUDM/slime#1013]
This is a followup after we have implemented  the basic and core features for FSDP within https://github.com/THUDM/slime/issues/74

For next chapter, we focus on training large scale MoE and hybrid model like GPT-OSS, Qwen MoE series and Qwen3-next to demonstrate FSDP's flexibility, and also, we will try to enhance our current functions and add more useful features!

So there some work we need to do:

## Model servering
- Qwen3 30BA3B https://github.com/THUDM/slime/issues/1060
- [x] GPT OSS https://github.com/THUDM/slime/pull/996
- Qwen3 next  https://github.com/THUDM/slime/issues/1061

## Parallelism / MoE Optimization
- Add expert parallelism https://github.com/THUDM/slime/issues/295
- Optimize context parallelism https://github.com/THUDM/slime/issues/1062
- Fused experts for better MoE training performance https://github.com/THUDM/slime/issues/1144

## Training-infer mismatch
- Add rollout routing replay(R3)
- Add MIS https://github.com/THUDM/slime/issues/1063

## Training features
- Learning rate scheduling https://github.com/THUDM/slime/issues/962
- Add LoRA support for training
- True on-policy MoE training
## Peformance Optimization
- Try to optimize memory usage for model like Qwen3 30BA3B https://github.com/THUDM/slime/issues/1060

## Doc
- Add FSDP usage doc: https://github.com/THUDM/slime/issues/1047

We try to make FSDP as a clean, easy-to-use training backend for slime, ***if you are interested in any part or want to develop your own features, please feel free to comment and take!*** let's work together to achieve this 🚀

**Comment by BaohaoLiao** [comment]
It would be great if the training scripts of the corresponding models could be realeased as examples. For example for GPT-OSS, how should we set the reasoning effort and so on.

**Comment by PopSoda2002** [comment]
> It would be great if the training scripts of the corresponding models could be realeased as examples. For example for GPT-OSS, how should we set the reasoning effort and so on.

Hi I think for GPT-OSS, I think we have already get a script which we can run on, checkout: https://github.com/THUDM/slime/pull/996,

And for other models, we are still supportingm, stay tuned!

**Comment by zhaochenyang20** [comment]
There are four remaining issues on the FSDP side; none of them are trivial. But worth trying. Hope you guys enjoy it.

#1060 
#1061 
#1062 
#1063
#295 


If you lack computing resources. Please contact Jiajun to access his approval. Welcome to the party.

Also, FSDP VLM still has something left to do. Welcome to contribute.

https://github.com/THUDM/slime/issues/1075

**Comment by mingMelody** [comment]
> There are four remaining issues on the FSDP side; none of them are trivial. But worth trying. Hope you guys enjoy it.
> 
> [#1060](https://github.com/THUDM/slime/issues/1060) [#1061](https://github.com/THUDM/slime/issues/1061) [#1062](https://github.com/THUDM/slime/issues/1062) [#1063](https://github.com/THUDM/slime/issues/1063) [#295](https://github.com/THUDM/slime/issues/295)
> 
> If you lack computing resources. Please contact Jiajun to access his approval. Welcome to the party.
> 
> Also, FSDP VLM still has something left to do. Welcome to contribute.
> 
> [#1075](https://github.com/THUDM/slime/issues/1075)

I'm ready to work on #1060. Could you please share how I can contact Jiajun to apply for computing resource approval?

**Comment by PopSoda2002** [comment]
> > There are four remaining issues on the FSDP side; none of them are trivial. But worth trying. Hope you guys enjoy it.
> > [#1060](https://github.com/THUDM/slime/issues/1060) [#1061](https://github.com/THUDM/slime/issues/1061) [#1062](https://github.com/THUDM/slime/issues/1062) [#1063](https://github.com/THUDM/slime/issues/1063) [#295](https://github.com/THUDM/slime/issues/295)
> > If you lack computing resources. Please contact Jiajun to access his approval. Welcome to the party.
> > Also, FSDP VLM still has something left to do. Welcome to contribute.
> > [#1075](https://github.com/THUDM/slime/issues/1075)
> 
> I'm ready to work on [#1060](https://github.com/THUDM/slime/issues/1060). Could you please share how I can contact Jiajun to apply for computing resource approval?

Hi, I think you can contact him joining the sglang slack and search Jiajun Li, does that make sense cc @guapisolo

**Comment by AceHao** [comment]
+1 on Lora FT support, for GLM5 GRPO
